Add organisations fields

steventux opened the pull request:
https://trello.com/c/Hf90xZZq/219-drop-down-for-organisations-in-eu-exit-statutory-instrument-specialist-publisher-document-type

Adds the abilities to select Primary publishing organisation and organisations for Statutory Instruments.

![screenshot from 2018-06-15 11-53-04](https://user-images.githubusercontent.com/93511/41464546-afb4720a-7092-11e8-96bb-fbad8e837708.png)

Commit:
Fetch Organisations from publishing API

Fetches all published organisations from the publishing API.
There are currently 1068 orgs so a pagination query is needed
to get them all.

Commit:
Rename Document#organisations to Document#schema_organisations

Document#organisations returns the static orgs from the
schema file. In order to use the links organisations effectively
we need to rename this so #organisations can be used for links.

This commit also adds a permissions check for '<document_type>_editor'
permissions so that permissions can be spread across organisations
for a specific document type."

Commit:
Add organisations options to statutory instrument form

Commit:
Make primary_publishing_organisation a selectable field

elliotcm commented on app/models/organisation.rb:
```diff
@@ -0,0 +1,42 @@
+        orgs << response["results"].map { |attrs| Organisation.new(attrs) }
+        if response["pages"] > current_page
+          current_page += 1
+        else
```
Will this enter an infinite loop if there's some issue with returning the orgs?

steventux replied:
It's a good question, I'd borrowed heavily from https://github.com/alphagov/specialist-publisher/blob/master/app/services/all_documents_finder.rb#L35 
I could put in some 'fuse' condition to stop after a finite number of iterations.

elliotcm commented on app/views/metadata_fields/_statutory_instruments.html.erb:
```diff
@@ -17,3 +17,29 @@
+        class: "select2 form-control",
+        multiple: true,
+        data: {
+          placeholder: "Select organisations"
```
I think the terms we wanted in the editor itself were "Publishing organisation" and "Other associated organisations".

elliotcm commented on app/models/statutory_instrument.rb:
```diff
@@ -12,16 +13,26 @@ class StatutoryInstrument < Document
-    "fef4ac7c-024a-4943-9f19-e85a8369a1f3"
+  def links
+    super.merge(
+      organisations: organisations,
```
Since `organisations` should also contain the primary publishing org we should probably union them here so we can provide the right list of orgs regardless of whether the user put the primary org in the additional field or not.

Commit:
Amend form labels

Commit:
Limit the number of potential requests to the publishing API

There are 1068 published organisations and we are querying with a page
size of 600 results per page. You do the math.
Ensures that an infinite loop isn't possible as a safeguard.

elliotcm reviewed (approved):
The explicit 10 loops makes me twitch a little because it's a time bomb for, what, 10 years from now, but the chances of us hitting 6k orgs seems slim so..  approved.

Commit:
Include publishing organisation with other organisations

The primary publishing organisation should be included in the organisations links.

This commit also removes the memoization of orgs from the publishing api
as the gds-api-adapters cache responses for 15 minutes.

elliotcm commented on app/models/statutory_instrument.rb:
```diff
@@ -27,7 +27,7 @@ def self.title
   def links
     super.merge(
-      organisations: organisations,
+      organisations: organisations << primary_publishing_organisation,
```
This will result in duplicates if they put the org in both fields, probably want `organisations | [primary_publishing_organisation]`
